pkg/kubelet/kubelet_resources.go

```diff
@@ -21,18 +21,21 @@ import (
 
 	"k8s.io/klog/v2"
 
-	"k8s.io/api/core/v1"
+	corev1 "k8s.io/api/core/v1"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	resourcehelper "k8s.io/component-helpers/resource"
 	"k8s.io/kubernetes/pkg/api/v1/resource"
+	kubefeatures "k8s.io/kubernetes/pkg/features"
 )
 
 // defaultPodLimitsForDownwardAPI copies the input pod, and optional container,
 // and applies default resource limits. it returns a copy of the input pod,
 // and a copy of the input container (if specified) with default limits
-// applied. if a container has no limit specified, it will default the limit to
-// the node allocatable.
-// TODO: if/when we have pod level resources, we need to update this function
-// to use those limits instead of node allocatable.
-func (kl *Kubelet) defaultPodLimitsForDownwardAPI(pod *v1.Pod, container *v1.Container) (*v1.Pod, *v1.Container, error) {
+// applied.
+// If a container has no limits specified, it defaults to the pod-level resources.
+// If neither container-level nor pod-level resources are specified, it defaults
```

Comment:
If neither container-level nor pod-level resources limits are specified, it defaults...

Reply:
Thanks! Done.